src/category_theory/is_connected.lean

```diff
@@ -56,39 +66,51 @@ This allows us to show that the functor X ⨯ - preserves connected limits.
 
 See https://stacks.math.columbia.edu/tag/002S
 -/
-class connected (J : Type v₂) [category.{v₁} J] extends inhabited J :=
-(iso_constant : Π {α : Type v₂} (F : J ⥤ discrete α), F ≅ (functor.const J).obj (F.obj default))
+class is_connected (J : Type v₂) [category.{v₁} J] extends is_connected_or_empty J : Prop :=
+[is_nonempty : nonempty J]
+
+attribute [instance, priority 100] is_connected.is_nonempty
 
 variables {J : Type v₂} [category.{v₁} J]
 
+/--
+If `J` is connected, any functor `F : J ⥤ discrete α` is isomorphic to
+the constant functor with value `F.obj j` (for any choice of `j`).
+-/
+def iso_constant [is_connected_or_empty J] {α : Type v₂} (F : J ⥤ discrete α) (j : J) :
```

Comment:
With a bit of pain, you could now remove the universe condition on `\alpha`, right? Not sure if it's worth it. But I can imagine that at some point you get handed a functor to `discrete nat` or `discrete bool` or something like that.

Reply:
Ok, I generalized to allow an arbitrary universe... I'm not sure that it is worth the pain, because now we need to write `.{v₁ v₂ v₃}` in a bunch of places. The diff is at https://github.com/leanprover-community/mathlib/pull/4136/commits/6cd0907164c4b269abafdef747572933eabcccb8, and if anyone sucks "yuck" I will revert it. :-)